overlord/configstate/configcore/runwithstate.go

```diff
@@ -116,6 +120,10 @@ func applyHandlers(dev sysconfig.Device, cfg config.Conf, handlers []configHandl
 			if !validCertOption(k) {
 				return fmt.Errorf("cannot set store ssl certificate under name %q: name must only contain word characters or a dash", k)
 			}
+		case strings.HasPrefix(k, "core.system.network.netplan."):
```

Comment:
same question about .netplan without the dot?

Reply:
of course though we need to reject netplanfoo

Reply:
Added and added missing tests for this.